plugin/src/helpers/config.ts

```diff
@@ -115,6 +129,39 @@ export async function checkConfig({ utils, netlifyConfig }): Promise<void> {
   }
 }
 
+async function getPreviouslyCopiedDatastoreFileName(publishDir: string) {
+  const files = await readdir(resolve(publishDir))
+  const match = files.filter(
+    (file) => file.startsWith('data-') && file.endsWith('.mdb'),
+  )
+
+  // eslint-disable-next-line unicorn/explicit-length-check
+  return match.length > 0 ? match[0] : null
+}
+/**
+ * Copies the contents of the Gatsby datastore file to the public directory in order
+ * to be uploaded to the CDN.
+ *
+ */
+export async function createMetadataFileAndCopyDatastore(
+  publishDir: string,
+  cacheDir: string,
+): Promise<void> {
+  const data = join(`${cacheDir}/data/datastore/data.mdb`)
+  const previousFileName = await getPreviouslyCopiedDatastoreFileName(
```

Comment:
I can definitely add a check for an existing `dataMetadata.json` file and using that filename alongside the check for the datastore file. 👍🏻 

Just to make sure I'm not missing context (as I'm understanding your comment as 'pick what you prefer for the approach but at least have logic for the metadata file')- is there any particular reason for why it might be preferable to only check for an existing `dataMetadata.json` file over the existence of the datastore file?

Reply:
I meant that, unless I'm misunderstanding, you're already persisting the name of the generated mdb file, so walking the filesystem to discover it shouldn't be needed as you can just read the filename from that metadata file.

Reply:
Ah ok, got it! It's not showing it as outdated because this file wasn't modified, but I've updated the code to check the metadata data file first, followed by the datastore file itself.